packages/eslint-rules/src/rules/prefer_amplify_errors.ts

```diff
@@ -0,0 +1,36 @@
+import { ESLintUtils } from '@typescript-eslint/utils';
+
+export const preferAmplifyErrorsRule = ESLintUtils.RuleCreator.withoutDocs({
+  create(context) {
+    return {
+      // This naming comes from @typescript-eslint/utils types.
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      NewExpression(node) {
+        const checkNode = (errorType: string) => {
+          if (
+            node.callee.type === 'Identifier' &&
+            node.callee.name === errorType
+          ) {
+            context.report({
+              messageId: 'useOfErrorDetected',
+              node,
+            });
+          }
+        };
+        checkNode('Error');
```

Comment:
nit. (feel free to ignore).
we could remove `checkNode` and inline this logic. 
In other places we have it because we're dealing with multiple symbols and messages.

Reply:
if it's unnecessary, I'll get rid of it